pkg/ipam-controller/controllers/node/node.go

```diff
@@ -57,12 +58,22 @@ func (r *NodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		reqLog.Info("node object removed")
 	}
 	// node updated, trigger sync for all pools
-	poolList := &ipamv1alpha1.IPPoolList{}
-	if err := r.Client.List(ctx, poolList, client.InNamespace(r.PoolsNamespace)); err != nil {
+	ipPoolList := &ipamv1alpha1.IPPoolList{}
+	if err := r.Client.List(ctx, ipPoolList, client.InNamespace(r.PoolsNamespace)); err != nil {
 		return ctrl.Result{}, err
 	}
-	for _, p := range poolList.Items {
-		r.NodeEventCh <- event.GenericEvent{
+	for _, p := range ipPoolList.Items {
+		r.IPPoolNodeEventCH <- event.GenericEvent{
+			Object: &ipamv1alpha1.IPPool{
+				ObjectMeta: metav1.ObjectMeta{Namespace: r.PoolsNamespace, Name: p.Name},
+			}}
+	}
+	cidrPoolList := &ipamv1alpha1.CIDRPoolList{}
+	if err := r.Client.List(ctx, cidrPoolList, client.InNamespace(r.PoolsNamespace)); err != nil {
+		return ctrl.Result{}, err
+	}
+	for _, p := range cidrPoolList.Items {
+		r.CIDRPoolNodeEventCH <- event.GenericEvent{
 			Object: &ipamv1alpha1.IPPool{
```

Comment:
You need to send out an event for `CIDRPool` right?

Reply:
good catch! Luckily this was not an issue in this case, because we use only name+namespace from the object. fixed